src/components/Header/Header.jsx

```diff
@@ -12,16 +14,46 @@ import {
 } from './Header.styled';
 
 export default function Header(props) {
+  const { pathname } = useLocation();
+  const { search } = useYoutubeData();
+  const [query, setQuery] = useState('Wizeline');
+
+  const doSearch = useCallback(() => {
+    search({ query });
+  }, [search, query]);
+
+  const onSearchInputChange = (e) => {
+    setQuery(e.target.value);
```

Comment:
This will call the API whenever we change the input, better call the API after we stop typing for, e.g 500ms, or after pressing `Enter` key

Reply:
Actually, this is the method to call the API, and it will only be triggered when the search form is submitted (in this case by pressing `Enter`). `onSearchInputChange ` only sets the internal state, not to handle API call.

https://github.com/tkhquang/react-certification-2021/pull/2/files/23f936aaaec503d65d3a9c3a19211aed9923b5c2#diff-b9b2b997906c030374ac927d0ae375be39da0e694f96304015c36b271f8902d6R29-R35